src/org/exist/indexing/StreamListener.java

```diff
@@ -43,6 +43,13 @@
         STORE,
 
         /**
+         * Mode for replacing the nodes of a document with another document
+         * this is really a group mode, which is later followed by {@link #REMOVE_ALL_NODES}
+         * and then {@link #STORE}
+         */
+        REPLACE_DOCUMENT,
```

Comment:
I do not see reason for it
```
setMode(REPLACE_DOCUMENT)
startReplaceDocument()

setMode(REMOVE_ALL_NODES)
startIndexDocument()
...
endIndexDocument()

setMode(STORE)
startIndexDocument()
...
endIndexDocument()

endReplaceDocument()```

You call `startReplaceDocument` at mode `REPLACE_DOCUMENT`, and `endReplaceDocument` at mode `STORE`.  That looks a bit strange as minimum.

Why you can't call `startReplaceDocument` at mode `UNKNOWN` (for example)?